Bucketeer2/Sources/Internal/Remote/ApiClientImpl.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+
+final class ApiClientImpl: ApiClient {
+
+    static let DEFAULT_REQUEST_TIMEOUT_MILLIS: Int64 = 30_000
+
+    private let endpoint: URL
+    private let apiKey: String
+    private let featureTag: String
+    private let session: Session
+    private let defaultRequestTimeoutMills: Int64
+    private let logger: Logger?
+
+    enum ResponseError: Error {
+        case unknown(URLResponse?)
+        case unacceptableCode(code: Int, response: ErrorResponse?)
+    }
+
+    init(endpoint: URL,
+         apiKey: String,
+         featureTag: String,
+         defaultRequestTimeoutMills: Int64 = ApiClientImpl.DEFAULT_REQUEST_TIMEOUT_MILLIS,
+         session: Session = URLSession.shared,
+         logger: Logger?) {
+
+        self.endpoint = endpoint
+        self.apiKey = apiKey
+        self.featureTag = featureTag
+        self.defaultRequestTimeoutMills = defaultRequestTimeoutMills
+        self.session = session
+        self.logger = logger
+        self.session.configuration.timeoutIntervalForRequest = TimeInterval(self.defaultRequestTimeoutMills)
+    }
+
+    func getEvaluations(user: JSON.User, userEvaluationsId: String, timeoutMillis: Int64?, completion: ((Result<GetEvaluationsResponse, Error>) -> Void)?) {
+        let startAt = Date()
+        let requestBody = GetEvaluationsRequestBody(
+            tag: self.featureTag,
+            user: user,
+            user_evaluations_id: userEvaluationsId,
+            source_id: .ios
+        )
+        let featureTag = self.featureTag
+        logger?.debug(message: "[API] Fetch Evaluation: \(requestBody)")
+        send(
+            requestBody: requestBody,
+            path: "v1/gateway/evaluations",
+            timeoutMillis: timeoutMillis,
+            completion: { (result: Result<(GetEvaluationsResponse, URLResponse), Error>) in
+                switch result {
+                case .success((var response, let urlResponse)):
+                    let endAt = Date()
+                    let duration = endAt.timeIntervalSince(startAt)
+                    response.seconds = duration
+                    let contentLength = urlResponse.expectedContentLength
+                    response.sizeByte = contentLength
+                    response.featureTag = featureTag
+                    completion?(.success(response))
+                case .failure(let error):
+                    completion?(.failure(error))
```

Comment:
Can you handle the error for `getEvaluations` and `registerEvents` and return it like this?
It will help when you implement the caller.

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/internal/remote/GetEvaluationsResult.kt#L15-L16

https://github.com/bucketeer-io/android-client-sdk/blob/main/bucketeer/src/main/kotlin/io/bucketeer/sdk/android/BKTException.kt

Reply:
OK!